cell_classification/segmentation_data_prep.py

```diff
@@ -209,18 +210,44 @@ def check_input(self):
         else:
             self.selected_markers = self.selected_markers
 
+            # check if selected markers are in conversion matrix
+            if not set(self.selected_markers).issubset(set(self.conversion_matrix.columns)):
+                raise ValueError("The selected markers are not in the conversion matrix")
+
+        # NORMALIZATION DICT
         # load or construct normalization dict
         if str(self.normalization_dict_path).endswith(".json"):
             self.normalization_dict = json.load(open(self.normalization_dict_path, "r"))
+
+            # check if selected markers are in normalization dict
+            if not set(self.selected_markers).issubset(set(self.normalization_dict.keys())):
+                raise ValueError("The selected markers are not in the normalization dict")
         else:
+            # function raises a generic FileNotFoundError if selected_marker file
+            # or segmentation_fname not in data_folders
             self.normalization_dict = self.calculate_normalization_matrix(
                 self.data_folders,
                 self.selected_markers,
             )
+        # check if normalization_quantile is in [0, 1]
+        if self.normalization_quantile < 0 or self.normalization_quantile > 1:
+            raise ValueError("The normalization_quantile is not in [0, 1]")
 
+        # CELL TYPE TABLE
         # load cell_types.csv
         self.cell_type_table = pd.read_csv(self.cell_table_path)
 
+        # check if cell_type_key is in cell_type_table
+        if self.cell_type_key not in self.cell_type_table.columns:
+            raise ValueError("The cell_type_key is not in the cell_type_table")
+        # check if segment_label_key is in cell_type_table
+        if self.segment_label_key not in self.cell_type_table.columns:
+            raise ValueError("The segment_label_key is not in the cell_type_table")
+
+        # check if sample_key is in cell_type_table
+        if self.sample_key not in self.cell_type_table.columns:
+            raise ValueError("The sample_key is not in the cell_type_table")
```

Comment:
We should check that all of the sample names in the cell table match the data_folder names

Reply:
So in the above three comments you write:
1. supply data_dir instead of data_folders, get the folders within `check_input`
2. check if unique cell types from cell table are present in the conversion matrix
3. check that all of the sample names in the cell table match data_folder names

I need to think about this more, but I think these could be overly restrictive. If we have a dataset where some FOVs should be excluded, 1. would be too restrictive and 3. would throw an exception. Also 2. could be better checked when it's needed in `get_marker_activity` and not here, since it could be that samples are excluded that have a certain cell type that is not in the conversion matrix. I worked mostly with super messy data and I just see that the more assumptions and the less degree of freedoms we'll have here, the more we need to change this stuff later.

Reply:
Agreed, lets skip these for now.